<launch>

<arg name="mav_name" default="bluebird"/>
  <arg name="object_name" default="$(arg mav_name)"/>
  <arg name="namespace" default="$(arg mav_name)" />

  <arg name="marker_frame" default="world" />

  <node name="vrpn_client_kinect" type="ros_vrpn_client" pkg="ros_vrpn_client" output="screen">
    <param name="vrpn_server_ip" value="vicon" />
    <param name="vrpn_coordinate_system" value="vicon" />
    <param name="timestamping_system" value="tracker" />
    <param name="object_name" value="kinectv2" />
    <param name="display_time_delay" value="true" />
    <param name="vicon_estimator/dt" value="0.01" />
    <param name="translational_estimator/kp" value="1.0" />
    <param name="translational_estimator/kv" value="0.1" />
    <param name="rotational_estimator/orientation_estimate_initial_covariance" value="1" />
    <param name="rotational_estimator/rate_estimate_initial_covariance" value="1" />
    <param name="rotational_estimator/orientation_process_covariance" value="0.00001" />
    <param name="rotational_estimator/rate_process_covariance" value="10" />
    <param name="rotational_estimator/orientation_measurementCovariance" value="0.005" />
    <param name="rotational_estimator/outlier_threshold_degrees" value="30" />
    <param name="rotational_estimator/maximum_outlier_count" value="50" />    
    <param name="rotational_estimator/output_minimal_quaternions" value="false" />
  </node>

  <node name="vrpn_client_chessboard" type="ros_vrpn_client" pkg="ros_vrpn_client" output="screen">
    <param name="vrpn_server_ip" value="vicon" />
    <param name="vrpn_coordinate_system" value="vicon" />
    <param name="timestamping_system" value="tracker" />
    <param name="object_name" value="chessboardv2" />
    <param name="display_time_delay" value="true" />
    <param name="vicon_estimator/dt" value="0.01" />
    <param name="translational_estimator/kp" value="1.0" />
    <param name="translational_estimator/kv" value="0.1" />
    <param name="rotational_estimator/orientation_estimate_initial_covariance" value="1" />
    <param name="rotational_estimator/rate_estimate_initial_covariance" value="1" />
    <param name="rotational_estimator/orientation_process_covariance" value="0.00001" />
    <param name="rotational_estimator/rate_process_covariance" value="10" />
    <param name="rotational_estimator/orientation_measurementCovariance" value="0.005" />
    <param name="rotational_estimator/outlier_threshold_degrees" value="30" />
    <param name="rotational_estimator/maximum_outlier_count" value="50" />    
    <param name="rotational_estimator/output_minimal_quaternions" value="false" />
  </node>

   <node pkg="vicon_depthcam_scan" name="image_converter" type="image_converter">
    </node>
    
    <!-- display 0: (default) not display / 1: display found corner -->
    <param name="display" type="int" value="1"/>
    <!-- maxboard -1: (default) infinite search board / 1<=: limit found number of boards -->
    <param name="maxboard" type="int" value="1"/>
    <!-- set frame_id if you want to use specific frame_id -->
    <!-- param name="frame_id" type="string" value="stereo_link"/-->
    <param name="rect0_size_x" type="double" value="0.07"/>
    <param name="rect0_size_y" type="double" value="0.07"/>
    <param name="grid0_size_x" type="int" value="6"/>
    <param name="grid0_size_y" type="int" value="7"/>
    <param name="type0" type="string" value="6x7chessboard"/>

    <node pkg="checkerboard_detector" type="checkerboard_detector"
          name="detector_node" respawn="false" >
      <remap from="camera_info" to="/camera/ir/camera_info"/>
      <remap from="image" to="/camera/ir/image_rect_conv"/>
    </node>

    <node pkg="vicon_depthcam_scan" type="transform_calc"
          name="transform_calc_node" output="screen">
    </node>

</launch>
